### PR TITLE
Bug fixes and more quality assurance

### LINKED ISSUE #8: Add a pre-release script to update static fallback data
Before every release, we need to generate and update the fallback data. This will introduce an automated approach and help on reducing maintenance effort.

### LINKED ISSUE #35: Extend cli test suite with file saving scenarios
When any of index.html or manifest.json file is provided to a cli, meta information is saved on them. Current functionality of that is not tested in any scenarios. We need to extend cli.test.js file by adding those test to cover existing functionality.

### LINKED ISSUE #69: Optimize build for node >=8.16.0
Remove tslib and change tsconfig target to align with engine >=8.16.0 req

### LINKED ISSUE #70: add favicon only flags
Here is the situation,
I want to generate a transparent `favicon`, and white background for `splash` and `icon`, but there is no such option.

when I set `{favicon: true}`, the program automatically generate `icon & splash` for me 

Thus here is my workaround, first I generate `favicon` with `{ opaque: false, iconOnly: true, favicon: true }` and then overwrite it with `{background:"#FFFFFF", favicon:false}` option.

I wonder if we can add `faviconOnly` flag.

### LINKED ISSUE #72: Utilize util.promisify()
As node engine req has been increased to >=8.16.0 with pptr v2 upgrade, we can use node's `util.promisify()`method on file service.

### LINKED ISSUE #73: Remove process.setMaxListeners call in puppets
As lib doesn't initialize a new browser instance for each screenshot anymore, there's no need to keep process.setMaxListeners() call in https://github.com/onderceylan/pwa-asset-generator/blob/master/src/helpers/puppets.ts#L311

### LINKED ISSUE #75: Error when icons key is missing from manifest json
**Describe the bug**

I get the following error if the "icons" key is missing from the webapp manifest.

    3:22:41 AM cli TypeError: Cannot read property 'filter' of undefined 

**To Reproduce**

Remove icons key in manifest file and generate assets with: 

    pwa-asset-generator ./frontend/assets/logo.svg ./build -i ./index.html -m ./manifest.json -b #000000

**System (please complete the following information):**
 - OS: [windows 10]
 - node version [v12.13.1]
 - cli version [2.1.2]

### LINKED ISSUE #77: Issue creating an output dir with recursive dir tree
When the output folder is non-existent, CLI creates a new folder for users. If a user provides a non-existent directory tree, it's not able to resolve and create the tree for the user.

### PR DESCRIPTION
**Pull request by onderceylan**
Hey guys, Santa 🎅 dropped by at my apartment tonight and left some commits in my socks, yay!

* Upgraded TypeScript to the latest version.
* Added visual regression tests and file update tests to improve quality assurance of the test suite. 
* Added a pre-release script to automatically update static fallback specs data before each release.
* Added a new CI workflow to check if the crawler script health which is used to generate static fallback specs data.
* Dropped tslib dependency as node engine is targeted over v8 after recent pptr upgrade.
* Refactored file service to utilize node's native util promisify - only possible when node engine is targeted over v8.
* Bug fixes and documentation improvements.

fixes #77 
fixes #75 
fixes #73 
fixes #72
fixes #70 
fixes #69 
fixes #35
fixes #8

### CODE REVIEW

File: src/main.test.ts [hunk 1]
```diff
@@ -1,88 +1,491 @@
...
+          background: 'coral',
+        },
+        'https://raw.githubusercontent.com/onderceylan/pwa-asset-generator/HEAD/static/logo.png',
+        `./temp/remote/${testSuite}`,
+      );
+
+      await assertMatchedSnapshots(result, testSuite);
+    });
+
+    test.skip('in svg format', async () => {
```
Comment (onderceylan) [inline src/main.test.ts]: Skipped tests have a dependency on static files hosted on master, I will enable them once this PR is merged.

File: tsconfig.json [hunk 2]
```diff
@@ -8,12 +8,12 @@
     "strict": true,
     "declaration": true,
     "forceConsistentCasingInFileNames": true,
-    "importHelpers": true,
```
Comment (onderceylan) [inline tsconfig.json]: This was related to tslib dep - keeps polyfills, and was used to be used to target ES5.

File: src/config/constants.ts [hunk 3]
```diff
@@ -209,117 +210,16 @@ export default {
...
-      landscape: { width: 1334, height: 750 },
-      scaleFactor: 2,
-    },
-    {
-      device: 'iPhone SE',
-      portrait: { width: 640, height: 1136 },
-      landscape: { width: 1136, height: 640 },
-      scaleFactor: 2,
-    },
-  ],
```
Comment (sneas) [inline src/config/constants.ts]: 👍for removing this JSON to a separate file.

**Pull request merged**